File: php/features.php
<?php
declare(strict_types=1);

// EmailValidationApi2 SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class EmailValidationApi2Features
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new EmailValidationApi2BaseFeature();
            case "test":
                return new EmailValidationApi2TestFeature();
            default:
                return new EmailValidationApi2BaseFeature();
        }
    }
}

File: py/features.py
# EmailValidationApi2 SDK feature factory

from feature.base_feature import EmailValidationApi2BaseFeature
from feature.test_feature import EmailValidationApi2TestFeature


def _make_feature(name):
    features = {
        "base": lambda: EmailValidationApi2BaseFeature(),
        "test": lambda: EmailValidationApi2TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
